sycl/source/detail/handler_impl.hpp

```diff
@@ -241,16 +223,16 @@ class handler_impl {
   // Allocation ptr to be freed asynchronously.
   void *MFreePtr = nullptr;
 
-  // Store information about the kernel arguments.
-  void *MKernelFuncPtr = nullptr;
+#ifndef __INTEL_PREVIEW_BREAKING_CHANGES
+  // TODO: remove in the next ABI-breaking window
+  // Today they are used only in the handler::setKernelNameBasedCachePtr
```

Comment:
This makes little sense to me. Everything under `source/detail/` is inside the `libsycl.so` and doesn't cross ABI boundary.

Reply:
These vars are used only when app is compiled with 6.3 compiler. In case of 6.3 compiler `StoreLambda` function calls `setKerneInfo()` that accept the compiler info from the integration header. After that `StoreLambda` calls `setDeviceKernelInfoPtr`. The problem with compatibility with 6.3 is that we need to store the data from integration header somewhere because `DeviceKernelInfoPtr` is not set yet.

You are right that this members does not cross ABI boundaries, but they are used only by the API that is under `#ifndef __INTEL_PREVIEW_BREAKING_CHANGES` macro. So I put them under preview, so that we don’t forget to remove them together with corresponding API in the next ABI-breaking window.